pkg/proxy/winkernel/proxier.go

```diff
@@ -1016,15 +1077,42 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 
 			if len(ep.hnsID) > 0 {
-				newHnsEndpoint, err = hns.getEndpointByID(ep.hnsID)
+				newHnsEndpoint = ep
 			}
 
 			if newHnsEndpoint == nil {
-				// First check if an endpoint resource exists for this IP, on the current host
-				// A Local endpoint could exist here already
-				// A remote endpoint was already created and proxy was restarted
 				newHnsEndpoint, err = hns.getEndpointByIpAddress(ep.ip, hnsNetworkName)
 			}
+
+			// Need to add an OutboundNat policy to local endpoints for DSR loopback to properly work.
+			if newHnsEndpoint != nil {
+				if newHnsEndpoint.isLocal && proxier.isDSR {
+					var addDSRPolicy = false
+					for _, po := range newHnsEndpoint.policies {
+						if po.Type == "OutBoundNAT" {
+							var outputSettings outboundnatpolicysetting
+							if err := json.Unmarshal([]byte(po.Settings), &outputSettings); err != nil {
+								klog.Errorf("Error Unmarshaling Endpoint Policy err: %v settings: %v", err, po.Settings)
+								continue
+							}
+							for _, des := range outputSettings.Destinations {
+								if des == newHnsEndpoint.ip {
+									addDSRPolicy = true
```

Comment:
this should remain false. addDSRPolicy implies that you need to add the policy

Reply:
Changing to "hasDSRPolicy". Bad form to set a flag to true.

Reply:
this is still not fixed?